devito/passes/clusters/aliases.py

```diff
@@ -133,7 +141,7 @@ def extractor(context):
         chosen, others = choose(exprs, aliases, selector)
         if not chosen:
             # Do not waste time
-            break
+            continue
```

Comment:
what does this change? I get it should be continue and don't want to exit the loop but wondering why wasn't breaking something before.

Reply:
no it's just that previously there was no point in keep going, while now the logic is different, and if the first round goes bad there's still hope for the second one

IOW, the previous version of the algo was acting constructively -- small temporaries first, then bigger ones. The newer version acts speculatively -- first it search for big ones, then smaller ones. This, for various reasons, makes everything so much simpler, while killing the OOB bug in an extremely natural way

Reply:
ok